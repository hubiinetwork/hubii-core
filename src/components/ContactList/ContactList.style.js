import { List } from 'antd';
import styled from 'styled-components';

export const StyledList = styled(List)`
  .ant-list-item-meta-title {
    color: ${({ theme }) => theme.palette.info3};
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.14rem;
  }
  .ant-list-item-meta-description {
    color: ${({ theme }) => theme.palette.secondary6};
    font-size: 0.86rem;
    font-weight: 400;
    line-height: 1rem;
  }
  .ant-btn-circle.ant-btn-sm,
  .ant-btn-circle-outline.ant-btn-sm {
    color: ${({ theme }) => theme.palette.secondary1};
    background: ${({ theme }) => theme.palette.secondary8};
    border-color: ${({ theme }) => theme.palette.secondary8};
  }
  .ant-list-item {
    border-bottom: 0.07rem solid ${({ theme }) => theme.palette.secondary7} !important;
    padding-bottom: 0.79rem;
  }
  em.ant-list-item-action-split {
    display: none;
  }
  .ant-btn.ant-btn-primary.ant-btn-circle.ant-btn-sm.ant-btn-icon-only {
    &:hover {
      color: ${({ theme }) => theme.palette.info};
      background: ${({ theme }) => theme.palette.secondary8} !important;
      border-color: ${({ theme }) => theme.palette.secondary8} !important;
    }
  }
  .ant-list-item-action {
    padding-top: 0.71rem;
  }
  .ant-list-item-action > li {
    padding: 0 0.29rem;
  }
`;
