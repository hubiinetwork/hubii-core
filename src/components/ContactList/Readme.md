Variants of ContactList:

```js
<div style={{ backgroundColor: '#26404d', padding: 25 }}>
  <ContactList
    data={[
      {
        name: 'Sample 1',
        address: '0xa7adde583f26c2f1c71e02f3bb729f0016033f76'
      },
      {
        name: 'Sample 2',
        address: '0xa7adde583f26c2f1c71e02f3bb729f0016033f77'
      },
      {
        name: 'Sample 3',
        address: '0xa7adde583f26c2f1c71e02f3bb729f0016033f78'
      }
    ]}
  />
</div>
```
