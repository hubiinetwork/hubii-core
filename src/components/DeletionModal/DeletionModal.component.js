import React from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  TextPrimary,
  Wrapper,
  StyledButton,
  ParentDiv,
  SecondaryHeader,
} from './DeletionModal.style';


/**
 * DeletionModal
 */
const DeletionModal = (props) => (
  <Wrapper>
    <Text> Are you sure you want to delete this {props.type}?</Text>
    <TextPrimary>
      <div>
        <SecondaryHeader>Name</SecondaryHeader> {props.name}
      </div>
      <div>
        <SecondaryHeader>Address</SecondaryHeader>{props.address}
      </div>
    </TextPrimary>
    <ParentDiv>
      <StyledButton type="danger" onClick={props.onDelete} id="delete">
        Delete
      </StyledButton>
      <StyledButton type="default" onClick={props.onCancel} id="cancel">
        Cancel
      </StyledButton>
    </ParentDiv>
  </Wrapper>
  );
DeletionModal.propTypes = {
  /**
   * Function to perform action when cancel button is clicked
   */
  onCancel: PropTypes.func,
  /**
   * Function to perform action when delete button is clicked
   */
  onDelete: PropTypes.func,
  /**
   * Name
   */
  name: PropTypes.string,
  /**
   * Address
   */
  address: PropTypes.string,
  /**
   * Type of Deletion (i.e "wallet", "contact")
   */
  type: PropTypes.string.isRequired,
};
export default DeletionModal;
