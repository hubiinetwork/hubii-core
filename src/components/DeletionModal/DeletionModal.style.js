import styled from 'styled-components';
import Button from 'components/ui/Button';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
export const Text = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: large;
  margin-bottom: 1.5rem;
`;

export const SecondaryHeader = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: large;
  margin: 0.5rem 0 0.5rem;
`;
export const TextPrimary = styled.div`
  color: ${({ theme }) => theme.palette.light};
  text-align: center;
`;
export const StyledButton = styled(Button)`
  min-width: 10.71rem;
  padding: 0.5rem 1rem;
  margin: 1rem;
`;
export const ParentDiv = styled.div`
  margin-top: 1.43rem;
  display: flex;
  justify-content: center;
`;
