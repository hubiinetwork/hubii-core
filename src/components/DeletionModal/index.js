export { default } from './DeletionModal.component';
