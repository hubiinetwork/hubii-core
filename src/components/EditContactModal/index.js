import EditContactModal from './EditContactModal.component';
export default EditContactModal;
