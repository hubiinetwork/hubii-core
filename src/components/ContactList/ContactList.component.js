import { List } from 'antd';
import * as React from 'react';
import PropTypes from 'prop-types';
import CopyToClipboard from 'react-copy-to-clipboard';
import DeletionModal from 'components/DeletionModal';
import EditContactModal from 'components/EditContactModal';

import StyledButton from '../ui/Button';
import { Modal } from '../ui/Modal';
import Notification from '../Notification';
import { StyledDiv, StyledList } from './ContactList.style';

/**
 * The ContactList Component shows list of contacts.
 */
export default class ContactList extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      modalVisibility: false,
      name: '',
      address: '',
      modalType: null,
    };
    this.showNotification = this.showNotification.bind(this);
    this.showModal = this.showModal.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
    this.onChange = this.onChange.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
  }

  onChange(input, type) {
    this.setState({
      [type]: input,
    });
  }

  showNotification() {
    const success = true;
    const message = 'Address copied to clipboard.';
    Notification(success, message);
  }

  showModal(item, modalType) {
    this.setState({
      modalVisibility: true,
      name: item.name,
      address: item.address,
      modalType,
    });
  }

  handleCancel() {
    this.setState({
      modalVisibility: false,
    });
  }

  handleDelete() {
    const { name, address } = this.state;
    this.setState({ modalVisibility: false });
    this.props.onDelete({ name, address });
  }

  handleEdit(oldContact) {
    const { onEdit } = this.props;
    const { name, address } = this.state;
    this.setState({ modalVisibility: false });
    onEdit({ name, address }, oldContact);
  }

  render() {
    const { size, layout, data } = this.props;
    const { name, address, modalType } = this.state;
    let modal;
    if (modalType === 'delete') {
      modal = (
        <DeletionModal
          name={name}
          address={address}
          onCancel={this.handleCancel}
          onDelete={this.handleDelete}
          type="contact"
        />
      );
    } else {
      modal = (
        <EditContactModal
          name={name}
          address={address}
          onEdit={(e) => this.handleEdit(e)}
          onChange={(input, type) => this.onChange(input, type)}
          contacts={data}
        />
      );
    }
    const Item = (item) => (
      <List.Item
        actions={[
          <StyledButton
            type="primary"
            shape="circle"
            icon="delete"
            size={'small'}
            onClick={() => this.showModal(item, 'delete')}
          />,
          <StyledButton
            type="primary"
            shape="circle"
            icon="edit"
            size={'small'}
            onClick={() => this.showModal(item, 'edit')}
          />,
          <CopyToClipboard text={item.address} key={2}>
            <StyledButton
              type="primary"
              shape="circle"
              icon="copy"
              size={'small'}
              onClick={this.showNotification}
            />
          </CopyToClipboard>,
        ]}
      >
        <List.Item.Meta title={item.name} description={item.address} />
      </List.Item>
    );
    return data.length > 0 ? (
      <div>
        <StyledList
          size={size}
          dataSource={data}
          renderItem={Item}
          itemLayout={layout}
        />
        <Modal
          footer={null}
          width={'585px'}
          maskClosable
          maskStyle={{ background: 'rgba(232,237,239,.65)' }}
          style={{ marginTop: '20px' }}
          visible={this.state.modalVisibility}
          onCancel={this.handleCancel}
          destroyOnClose
        >
          {modal}
        </Modal>
      </div>
    ) : (
      <StyledDiv>{this.props.message}</StyledDiv>
    );
  }
}

ContactList.defaultProps = {
  size: 'small',
  layout: 'horizontal',
  message: 'There are no contacts added yet.',
};

ContactList.propTypes = {
  /**
   * Array of contacts whose list is to be shown.
   */
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  /**
   * Antd prop layout of list
   */
  layout: PropTypes.string,
  /**
   * Message to shown if list is empty
   */
  message: PropTypes.string,
  /**
   * size of antd list component
   */
  size: PropTypes.oneOf(['default', 'small', 'large']),
  /**
   * Function to be executed when edit button of modal is pressed
   */
  onEdit: PropTypes.func.isRequired,
  /**
   * Function to be executed when delete button of modal is pressed
   */
  onDelete: PropTypes.func.isRequired,
};
