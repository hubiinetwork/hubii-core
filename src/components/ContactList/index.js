import ContactList from './ContactList.component';
export default ContactList;
