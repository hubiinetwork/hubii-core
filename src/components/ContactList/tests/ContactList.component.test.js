import React from 'react';
import { shallow } from 'enzyme';
import ContactList from 'components/ContactList';
import DeletionModal from 'components/DeletionModal';
import EditContactModal from 'components/EditContactModal';

describe('<ContactList/>', () => {
  const contacts = [
    {
      name: 'mike',
      address: '0x3123123',
    },
    {
      name: 'joe',
      address: '0x123123123',
    },
  ];
  const onDelete = jest.fn();
  const onEdit = jest.fn();

  describe('shallow mount', () => {
    it('should render <ContactList/> with data', () => {
      const shallowWrapper = shallow(
        <ContactList
          data={contacts}
          onEdit={onEdit}
          onDelete={onDelete}
        />
      );
      expect(shallowWrapper).toMatchSnapshot();
    });

    it('should render <ContactList/> without data', () => {
      const shallowWrapper = shallow(
        <ContactList
          data={[]}
          onEdit={onEdit}
          onDelete={onDelete}
        />
      );
      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('Modal functionality', () => {
    let wrapper;
    let contactList;
    beforeEach(() => {
      wrapper = shallow(
        <ContactList
          data={contacts}
          onEdit={onEdit}
          onDelete={onDelete}
        />
      );
      contactList = wrapper.instance();
    });
    describe('showModal functionality', () => {
      it('should execute the showModal function and change the state', () => {
        const item = {
          name: 'mike',
          address: '0x1231231',
        };
        const modalType = 'edit';
        contactList.showModal(item, modalType);
        expect(contactList.state.name).toEqual(item.name);
        expect(contactList.state.address).toEqual(item.address);
        expect(contactList.state.modalType).toEqual(modalType);
        expect(contactList.state.modalVisibility).toEqual(true);
      });

      it('when the modalType is delete <ContactDeletionModal/> should be rendered', () => {
        const modalType = 'delete';
        wrapper.setState({ modalType });
        expect(wrapper.find(DeletionModal).length).toEqual(1);
      });

      it('when the modalType is anything else <EditContactModal/> should be rendered', () => {
        const modalType = '';
        wrapper.setState({ modalType });
        expect(wrapper.find(EditContactModal).length).toEqual(1);
      });
    });

    it('should execute the handleCancel function', () => {
      contactList.handleCancel();
      expect(contactList.state.modalVisibility).toBe(false);
    });

    it('should execute the handleDelete function', () => {
      contactList.handleDelete();
      expect(contactList.state.modalVisibility).toBe(false);
      expect(contactList.props.onDelete).toHaveBeenCalledTimes(1);
    });

    it('should execute the onChange function', () => {
      const input = 'mike';
      const type = 'address';
      contactList.onChange(input, type);
      expect(contactList.state[type]).toEqual(input);
    });

    it('should execute the showNotification function', () => {
      const showNotifiction = jest.spyOn(contactList, 'showNotification');
      contactList.showNotification();
      expect(showNotifiction).toHaveBeenCalledTimes(1);
    });
  });
});
