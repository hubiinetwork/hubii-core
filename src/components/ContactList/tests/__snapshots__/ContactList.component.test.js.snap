// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactList/> shallow mount should render <ContactList/> with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactList
    data={
        Array [
            Object {
              "address": "0x3123123",
              "name": "mike",
            },
            Object {
              "address": "0x123123123",
              "name": "joe",
            },
          ]
    }
    layout="horizontal"
    message="There are no contacts added yet."
    onDelete={[Function]}
    onEdit={[Function]}
    size="small"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ContactList.style__StyledList
          dataSource={
                    Array [
                              Object {
                                "address": "0x3123123",
                                "name": "mike",
                              },
                              Object {
                                "address": "0x123123123",
                                "name": "joe",
                              },
                            ]
          }
          itemLayout="horizontal"
          renderItem={[Function]}
          size="small"
/>,
        <Modal.style
          destroyOnClose={true}
          footer={null}
          maskClosable={true}
          maskStyle={
                    Object {
                              "background": "rgba(232,237,239,.65)",
                            }
          }
          onCancel={[Function]}
          style={
                    Object {
                              "marginTop": "20px",
                            }
          }
          visible={false}
          width="585px"
>
          <Form(EditContactModal)
                    address=""
                    name=""
                    onChange={[Function]}
                    onEdit={[Function]}
                    validateEdit={[Function]}
          />
</Modal.style>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dataSource": Array [
            Object {
              "address": "0x3123123",
              "name": "mike",
            },
            Object {
              "address": "0x123123123",
              "name": "joe",
            },
          ],
          "itemLayout": "horizontal",
          "renderItem": [Function],
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Form(EditContactModal)
            address=""
            name=""
            onChange={[Function]}
            onEdit={[Function]}
            validateEdit={[Function]}
/>,
          "destroyOnClose": true,
          "footer": null,
          "maskClosable": true,
          "maskStyle": Object {
            "background": "rgba(232,237,239,.65)",
          },
          "onCancel": [Function],
          "style": Object {
            "marginTop": "20px",
          },
          "visible": false,
          "width": "585px",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "address": "",
            "name": "",
            "onChange": [Function],
            "onEdit": [Function],
            "validateEdit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ContactList.style__StyledList
            dataSource={
                        Array [
                                    Object {
                                      "address": "0x3123123",
                                      "name": "mike",
                                    },
                                    Object {
                                      "address": "0x123123123",
                                      "name": "joe",
                                    },
                                  ]
            }
            itemLayout="horizontal"
            renderItem={[Function]}
            size="small"
/>,
          <Modal.style
            destroyOnClose={true}
            footer={null}
            maskClosable={true}
            maskStyle={
                        Object {
                                    "background": "rgba(232,237,239,.65)",
                                  }
            }
            onCancel={[Function]}
            style={
                        Object {
                                    "marginTop": "20px",
                                  }
            }
            visible={false}
            width="585px"
>
            <Form(EditContactModal)
                        address=""
                        name=""
                        onChange={[Function]}
                        onEdit={[Function]}
                        validateEdit={[Function]}
            />
</Modal.style>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dataSource": Array [
              Object {
                "address": "0x3123123",
                "name": "mike",
              },
              Object {
                "address": "0x123123123",
                "name": "joe",
              },
            ],
            "itemLayout": "horizontal",
            "renderItem": [Function],
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Form(EditContactModal)
              address=""
              name=""
              onChange={[Function]}
              onEdit={[Function]}
              validateEdit={[Function]}
/>,
            "destroyOnClose": true,
            "footer": null,
            "maskClosable": true,
            "maskStyle": Object {
              "background": "rgba(232,237,239,.65)",
            },
            "onCancel": [Function],
            "style": Object {
              "marginTop": "20px",
            },
            "visible": false,
            "width": "585px",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "address": "",
              "name": "",
              "onChange": [Function],
              "onEdit": [Function],
              "validateEdit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ContactList/> shallow mount should render <ContactList/> without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactList
    data={Array []}
    layout="horizontal"
    message="There are no contacts added yet."
    onDelete={[Function]}
    onEdit={[Function]}
    size="small"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "There are no contacts added yet.",
    },
    "ref": null,
    "rendered": "There are no contacts added yet.",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "There are no contacts added yet.",
      },
      "ref": null,
      "rendered": "There are no contacts added yet.",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
